modelforge/potential/aimnet2.py

```diff
@@ -0,0 +1,750 @@
+            dtype=vector_prot_step2.dtype,
+        )  # Shape: (num_of_atoms, 3)
+
+        # Accumulate the vector contributions using index_add_
+        vector_contributions.index_add_(0, idx_j, vector_prot_step2)
+
+        # Step 3: Compute the Euclidean Norm for each atom
+        vector_norms = torch.norm(
+            vector_contributions, p=2, dim=1
+        )  # Shape: (num_of_atoms,)
+
+        return radial_contributions, vector_norms
+
+    def forward(
+        self,
+        atomic_embedding: torch.Tensor,
+        partial_charges: torch.Tensor,
+        pair_indices: torch.Tensor,
+        f_ij_cutoff: torch.Tensor,
+        r_ij: torch.Tensor,
+    ) -> torch.Tensor:
+        """
+        Forward pass of the message module.
+
+        Parameters
+        ----------
+        atomic_embedding : torch.Tensor
+            The embedding of each atom.
+        partial_charges : torch.Tensor
+            The partial charges of each atom.
+        pair_indices : torch.Tensor
+            The list of atom pairs.
+        f_ij_cutoff : torch.Tensor
+            The cutoff function applied to the radial symmetry functions.
+        r_ij : torch.Tensor
+            The displacement vectors between atom pairs.
+
+        Returns
+        -------
+        torch.Tensor
+            Updated atomic embeddings and partial charges.
+        """
+
+        # Step 1: Calculate radial and vector contributions for atomic embeddings (Equation 4 and 5)
+        radial_contributions_emb, vector_contributions_emb = (
+            self.calculate_contributions(
+                atomic_embedding,
+                pair_indices,
+                f_ij_cutoff,
+                r_ij,
+                use_charge_layer=False,
+            )
+        )
+
+        if not self.is_first_module:
+            # For subsequent message modules, calculate contributions for charges too
+            radial_contributions_charge, vector_contributions_charge = (
+                self.calculate_contributions(
+                    partial_charges,
+                    pair_indices,
+                    f_ij_cutoff,
+                    r_ij,
+                    use_charge_layer=True,
+                )
+            )
+
+            # Combine contributions
+            feature_vector_emb = torch.cat(
+                [radial_contributions_emb, vector_contributions_emb.unsqueeze(1)], dim=1
+            )
+            feature_vector_charge = torch.cat(
+                [radial_contributions_charge, vector_contributions_charge.unsqueeze(1)],
+                dim=1,
+            )
+
+            return torch.cat([feature_vector_emb, feature_vector_charge], dim=1)
+
+        # For the first message module, only return the atomic embedding contributions
+        feature_vector = torch.cat(
+            [radial_contributions_emb, vector_contributions_emb.unsqueeze(1)], dim=1
+        )
+        return feature_vector
+
+
+# class BaseMessageModule(nn.Module):
```

Comment:
Can we remove this block of commented out code?

Reply:
removed the block